app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/ExoPlayerAdapter.java

```diff
@@ -0,0 +1,343 @@
+package android.support.v17.leanback.supportleanbackshowcase.app.media;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.net.Uri;
+import android.os.Handler;
+import android.support.v17.leanback.media.PlaybackGlueHost;
+import android.support.v17.leanback.media.PlayerAdapter;
+import android.support.v17.leanback.media.SurfaceHolderGlueHost;
+import android.support.v17.leanback.supportleanbackshowcase.R;
+import android.view.SurfaceHolder;
+
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.DefaultLoadControl;
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.ExoPlayerFactory;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.extractor.DefaultExtractorsFactory;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.util.Util;
+
+/**
+ * This implementation extends the {@link PlayerAdapter} with a {@link SimpleExoPlayer}.
+ */
+public class ExoPlayerAdapter extends PlayerAdapter implements ExoPlayer.EventListener{
+
+    Context mContext;
+    final SimpleExoPlayer mPlayer;
+    SurfaceHolderGlueHost mSurfaceHolderGlueHost;
+    final Runnable mRunnable = new Runnable() {
+        @Override
+        public void run() {
+            getCallback().onCurrentPositionChanged(ExoPlayerAdapter.this);
+            getCallback().onBufferedPositionChanged(ExoPlayerAdapter.this);
+            mHandler.postDelayed(this, getUpdatePeriod());
+        }
+    };
+    final Handler mHandler = new Handler();
+    boolean mInitialized = false;
+    Uri mMediaSourceUri = null;
+    boolean mHasDisplay;
+    boolean mBufferingStart;
+    @C.StreamType int mAudioStreamType;
```

Comment:
@C.StreamType is an annotation in Exoplayer

https://google.github.io/ExoPlayer/doc/reference/index.html?com/google/android/exoplayer2/C.StreamType.html